services/venice-router/src/main/java/com/linkedin/venice/router/RouterServer.java

```diff
@@ -318,6 +324,13 @@ public RouterServer(
         config.getRefreshAttemptsForZkReconnect(),
         config.getRefreshIntervalForZkReconnectInMs());
     this.liveInstanceMonitor = new HelixLiveInstanceMonitor(this.zkClient, config.getClusterName());
+
+    D2Client d2Client = D2ClientFactory.getD2Client(config.getZkConnection(), Optional.empty());
```

Comment:
This is not right as `getZkConnection` is for Venice ZK.
I think we should let the backend pass an instance of `ZKClient` instead of constructing it here.

Reply:
@gaojieliu 

I don't believe we need an instance of `ZKClient` here. We'd need the [ZK Address](https://jarvis.corp.linkedin.com/codesearch/result/?name=VeniceRouterConfig.java&path=venice%2Fservices%2Fvenice-router%2Fsrc%2Fmain%2Fjava%2Fcom%2Flinkedin%2Fvenice%2Frouter&reponame=linkedin%2Fvenice#406) which is coming from the [router config](https://jarvis.corp.linkedin.com/codesearch/result/?name=RouterServer.java&path=venice%2Fservices%2Fvenice-router%2Fsrc%2Fmain%2Fjava%2Fcom%2Flinkedin%2Fvenice%2Frouter&reponame=linkedin%2Fvenice#207) file.   

If we can't use the `getZKConnection`, is there another way to create an instance of `D2Client`?

Reply:
I saw the code got checked in and the zk zddress in `router config` file is for Venice Zookeeper, which is different from D2 Zookeeper, and I assume here we want to use D2 Zookeeper Address to create a D2 client to access push status store, right?
The idea in the original comment is that it is easy to get a D2 client from the internal factory, and if that is not easy for testing, we will need to introduce an additional config to hold D2 Zookeeper address.